### MINI PROJECT

In [3]:
import numpy as np 
import pandas as pd
pd.plotting.register_matplotlib_converters()
import seaborn as sns
import matplotlib.pyplot as plt 
%matplotlib inline
print("Setup Complete")


Setup Complete


In [4]:
transfusion = pd.read_csv(r"C:\Users\USER\Downloads\archive\transfusion.csv")
transfusion

Recency (months)  Frequency (times)  Monetary (c.c. blood)  \
0                   2                 50                  12500   
1                   0                 13                   3250   
2                   1                 16                   4000   
3                   2                 20                   5000   
4                   1                 24                   6000   
..                ...                ...                    ...   
743                23                  2                    500   
744                21                  2                    500   
745                23                  3                    750   
746                39                  1                    250   
747                72                  1                    250   

     Time (months)  whether he/she donated blood in March 2007  
0               98                                           1  
1               28                                           1  
2               35                                           1  
3               45                                           1  
4               77                                           0  
..             ...                                         ...  
743             38                                           0  
744             52                                           0  
745             62                                           0  
746             39                                           0  
747             72                                           0  

[748 rows x 5 columns]

R (Recency - months since last donation), F (Frequency - total number of donation), M (Monetary - total blood donated in c.c.), T (Time - months since first donation), and a binary variable representing whether he/she donated blood in March 2007 (1 stand for donating blood; 0 stands for not donating blood).

In [5]:
transfusion.head(5)

Recency (months)  Frequency (times)  Monetary (c.c. blood)  Time (months)  \
0                 2                 50                  12500             98   
1                 0                 13                   3250             28   
2                 1                 16                   4000             35   
3                 2                 20                   5000             45   
4                 1                 24                   6000             77   

   whether he/she donated blood in March 2007  
0                                           1  
1                                           1  
2                                           1  
3                                           1  
4                                           0

In [6]:
transfusion.tail(5)

Recency (months)  Frequency (times)  Monetary (c.c. blood)  \
743                23                  2                    500   
744                21                  2                    500   
745                23                  3                    750   
746                39                  1                    250   
747                72                  1                    250   

     Time (months)  whether he/she donated blood in March 2007  
743             38                                           0  
744             52                                           0  
745             62                                           0  
746             39                                           0  
747             72                                           0

In [7]:
#getting a random sample of the dataset
transfusion.sample(n=6)

Recency (months)  Frequency (times)  Monetary (c.c. blood)  \
223                 4                  1                    250   
710                23                  2                    500   
381                14                  1                    250   
538                 2                  8                   2000   
362                 4                  3                    750   
733                23                  1                    250   

     Time (months)  whether he/she donated blood in March 2007  
223              4                                           0  
710             23                                           0  
381             14                                           0  
538             38                                           1  
362             71                                           0  
733             23                                           0

In [8]:
# checking for missing values
transfusion.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 748 entries, 0 to 747
Data columns (total 5 columns):
 #   Column                                      Non-Null Count  Dtype
---  ------                                      --------------  -----
 0   Recency (months)                            748 non-null    int64
 1   Frequency (times)                           748 non-null    int64
 2   Monetary (c.c. blood)                       748 non-null    int64
 3   Time (months)                               748 non-null    int64
 4   whether he/she donated blood in March 2007  748 non-null    int64
dtypes: int64(5)
memory usage: 29.3 KB


In [10]:
# to detect missing values 
transfusion.isna().sum()

Recency (months)                              0
Frequency (times)                             0
Monetary (c.c. blood)                         0
Time (months)                                 0
whether he/she donated blood in March 2007    0
dtype: int64

Table 1. Descriptive statistics of the data

Variable Data Type Measurement Description min max mean std
Recency quantitative Months Input 0.03 74.4 9.74 8.07 Frequency quantitative Times Input 1 50 5.51 5.84
Monetary quantitative c.c. blood Input 250 12500 1378.68 1459.83
Time quantitative Months Input 2.27 98.3 34.42 24.32
Whether he/she donated blood in March 2007 binary 1=yes 0=no Output 0 1 1 (24%) 0 (76%)


In [20]:
transfusion['Recency (months)'].mean()

9.506684491978609

In [32]:
transfusion.groupby["whether he/she donated blood in March 2007"]

TypeError: 'method' object is not subscriptable